src/github3/pulls.py

```diff
@@ -575,6 +576,21 @@ def reviews(self, number=-1, etag=None):
         url = self._build_url('reviews', base_url=self._api)
         return self._iter(int(number), url, PullReview, etag=etag)
 
+    def statuses(self, number=-1, etag=None):
+        """Iterate over the statuses associated with this pull request.
+
+        :param int number:
+            (optional), number of statuses to return. Default: -1 returns all
+            available statuses.
+        :param str etag:
+            (optional), ETag from a previous request to the same endpoint
+        :returns:
+            generator of statuses for this pull request
+        :rtype:
+            :class:`~github3.repos.Status`
+        """
+        return self._iter(int(number), self.statuses_url, status.Status,  etag=etag)
```

Comment:
Everywhere else we build a URL, (search for `_build_url` in this file for example) but here we're using the URL returned by GitHub. In the past they've been flat out wrong. GitHub recommended to me to just build URLs from scratch. That says to me that we should take a look at what this looks like now, and try to regenerate it from `self._api`.

Reply:
Can you help me understand how `_build_url` is used pleae?

 For the Status endpoint which looks like `/repos/:owner/:repo/statuses/:ref`, I need the base url `/repos/:owner/:repo/` and the ref which i can get from the `Head.sha`. However, I am unsure how to get that base url within the `PullRequest` context as `self._api` there looks like `https://api.github.com/repos/octocat/Hello-World/pulls/1347`

Reply:
So you could do

```py
self.repository._api
```

To get the base url you need, so then you would construct the call like so:

```py
self._build_url('statuses', self.head.sha, base_url=self.repository._api)
```

tl;dr, `_build_url` takes positional args as parts of the path to be added and `base_url` is an optional keyword argument that defaults to `https://api.github.com` unless specified otherwise.

Reply:
ThanKs for the explanation. I mostly wasn't sure on how to get the base url in this case, so thanks I can see that working. 

It builds the repo url from the `PullDestination` login and name fields using `ShortRepository`. I think it would be `self.repository.respository._api` though right (PullRequest.PullDestination.ShortRespository._api)? 

It seems roundabout but if we can't always trust the returned URL as you say, then this makes sense.

Reply:
> It builds the repo url from the PullDestination login and name fields using ShortRepository. I think it would be self.repository.respository._api though right (PullRequest.PullDestination.ShortRespository._api)?

I was looking at PullDestination's docstring :man_facepalming: so yes, that sounds fine. I think, though, that it would be `self.base.repository` if and only if `self.base` is actually present, kind of like https://github.com/sigmavirus24/github3.py/blob/f598e02ea8b89544c265923239825eea82f135e2/src/github3/pulls.py#L229

Reply:
Nope, you were right idk what i was looking at lol. my bad. I guess I'll return `[]` in the case there is no base, but I'm confused how that's a valid case?